modules/optimization/doc/content/source/reporters/VectorSum.md

```diff
@@ -0,0 +1,30 @@
+# VectorSum
+
+!syntax description /Reporters/VectorSum
+
+## Overview
+
+`VectorSum` operates on the elements of a vector in a reporter using a parsed function containing only two variables, `vi` and `vplus`.
```

Comment:
@giud @dschwen Would you recommend some different wording for this description and comment on the implementation of this object?  It is doing the vector reduction we talked about using a parsed function to define the function that will be used to accumulate the result as it iterates through the components of a vector.  It is doing what I want but I needs better naming.  The functions are also kind of awkward.  The class name is also bad.

Reply:
ParsedVectorReduction ? Or ParsedRealStdVectorReduction ?
it s going from std::vector to 1 Real right?

Reply:
Yes, this one goes from a vector to a scalar and the vectorOfvector goes from a matrix to a vector.

Reply:
ParsedStdVectorRealReductionReporter would work?
Parsed -> parsed expression
Reduction -> for the dimensionality reduction
StdVectorReal -> only type it supports?
Reporter -> needed because the same operation on an array auxvariable to a regular field one would be (roughly) the same name otherwise